test/types/type-provider.test-d.ts

```diff
@@ -437,3 +436,74 @@ expectAssignable(server.withTypeProvider<JsonSchemaToTsProvider>().get<{Reply: b
     return true
   }
 ))
+
+// -------------------------------------------------------------------
+// FastifyPlugin: Auxiliary
+// -------------------------------------------------------------------
+
+interface AuxiliaryPluginProvider extends FastifyTypeProvider { output: 'plugin-auxiliary' }
```

Comment:
What is the meaning of `{ output: 'plugin-auxiliary' }`, I've never seen this syntax after an `extends`

Reply:
> What is the meaning of { output: 'plugin-auxiliary' }, I've never seen this syntax after an extends

Ah, that just maps all parameters (`body`, `params`, `query` and `headers`) to infer as `type T = 'plugin-auxiliary'` for that provider. I tend to find literal strings a little bit easier read as vscode highlights string literals in editor. Also TypeScript treats string literals and `const` boolean and numbers are as distinct (ensuring there's no potential type overlap), but strings generally stand out more :D

![image](https://user-images.githubusercontent.com/3048342/178098751-f73cd48e-95eb-42d2-a7d6-65c5f37f31c8.png)

Reply:
Should we document this "trick" in our Typescript section?

Reply:
@mcollina Yes I think so. Speaking of which, I actually happened upon this project the other day (https://github.com/turkerdev/fastify-type-provider-zod/blob/main/src/index.ts#L5-L7) which allows Zod (a VERY popular inference package for TypeScript) to be integrated into Fastify. There's actually a whole ecosystem of these inference / validation libraries out there now (`io-ts`, `myzod`, `runtypes`, and even `yup`) all of which should be compatible with Type Providers. So documentation to help integrators may be a good idea.

Also, I found the Zod project to be an excellent example of integrating types, validation and serialization (and actually prompted the `TypeCompiler` TypeBox implementation. So I'd imagine the documentation could also include information on setting up `setValidatorCompiler()` and `setSerializerCompiler()` infrastructure as well as general information on extending the `FastifyTypeProvider` interface.

It would also be good to include the Zod Type Provider in the documentation too :)

Reply:
For future reference [YupTypeProvider](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzhARgKwKYGMYBo4DOMUwAdgOb6kCuIKGU+AJgIYwb4CSpAZgwCoBPMBzgBBUoIDKWABYYQLOAF84PKBBBwA5IOphtAWABQoSLDgAxFkWA9B+JNdv2hIgAoaAbsCYMVahpa2jw2MHaCRsYmGAAe5vBk7FChWBhwAJr6bhieED5+UHBx7KRMBFZhETl5Bf4IJnBNyNQwYK0AXHAwssAEANraZO0w2gC6xbGl5eKSMvKKcAD8cNx8UDkAPD19g8Ot4wB8cF3UpADWpBAA7qQmyiYmWBCkRGpV9nAAvJUuggAUAEoAHTXYA9GreXwMTZZMCQ-LQqCHIHAyBEf7aAD02kcjWaBDkChYXQaxmaFLgKAgTEEXVQmBw-zJlNZcFILBAGC6RBIFFRUAwAEdqMBBUwgbh8WzmixyNz2bR6FABcLReLUejwcAvBhUUkMPKVYCpeSZc1icAADY84hkcioy1WyXS83XDAoAjghW8+2o6hQZ0gmhWq0sFBWvUm10yrCCtgYJgAeVIXVY7FRflC1CtMH+QO+x1IGGucAAIgmgdGzazlIDXQ9jMp8P9BUL8IKCIDC4hXc9XvAkByufg5aInfh3Z7vfg4xgE8nSAEfm3gdTafd68YgA)